### PR TITLE
[BUG] Collection transformer wrapper bugfixing and `PlateauFinder` refactor

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
#709 made some changes to the collection wrapper due to test failures. Fixes those.

Moves `PlateauFinder` to its own file.

### CODE REVIEW

File: aeon/transformations/segment.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+"""Transformations for segmenting time series."""
```
Comment (TonyBagnall) [inline aeon/transformations/segment.py:1]: looks like it can move to segmentation to me (in a different PR)

**Review by TonyBagnall — DISMISSED** [review]
We can deprecate and move PlateauFinder once BaseSegmenter is in

**Comment by MatthewMiddlehurst** [comment]
Agreed, reverted the changes so this is just the wrapper.

File: aeon/transformations/collection/_collection_wrapper.py [hunk 2]
```diff
@@ -47,7 +50,7 @@ class CollectionToSeriesWrapper(BaseTransformer):
     def __init__(
         self,
         transformer,
-        _output_convert=None,
+        _output_convert="auto",
```
Comment (TonyBagnall) [inline aeon/transformations/collection/_collection_wrapper.py:53]: I think its better not to do the auto conversion myself. I think at least it should be a parameter so the user can turn it off, you can default to auto if you want

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**